application/src/main/kotlin/net/corda/v5/application/crypto/DigitalSignatureVerificationService.kt

```diff
@@ -13,17 +15,18 @@ import java.security.PublicKey
 @DoNotImplement
 interface DigitalSignatureVerificationService {
     /**
-     * Verifies a digital signature by using [signatureSpec].
-     * Always throws an exception if verification fails.
      *
-     * @param publicKey The signer's [PublicKey].
-     * @param signatureSpec The signature spec.
-     * @param signatureData The signatureData on a message.
-     * @param clearData The clear data/message that was signed (usually the Merkle root).
+     * Check a payload and signature, and optionally a digest algorithm, check the signature matches using a specific public key.
+     * The signature spec including digest will be extracted from the payload. This will either return cleanly, in which the case
+     * the verified signatures are returned, or exceptions will be thrown
+     *
+     * @param payload The payload/message that may be in clear data that was signed (e.g. the Merkle root).
+     * @param publicKeys A set of verified public keys to use for verification. Some of the signatures in signed document must have keys in this set.
+     * @return The metadata associated with the signature, passed in to [DigitalSigningService.sign]
      *
      * @throws CryptoSignatureException If verification of the digital signature fails.
      * @throws IllegalArgumentException If the signature scheme is not supported or if any of the clear or signature
      * data is empty.
      */
-    fun verify(publicKey: PublicKey, signatureSpec: SignatureSpec, signatureData: ByteArray, clearData: ByteArray)
+    fun <T, M> verify(payload: DigitalSignedDocument<T, M>, publicKeys: Set<PublicKey>): List<DigitalSignature>
```

Comment:
think this should be `document`, `signedDocument` or `digitalSignedDocument` instead of `payload`.